### ISSUE TITLE
Model from the best epoch should not be used for prediction when patience is set to None.

### ISSUE THREAD

**Issue by enwq** [WenjieDu/PyPOTS#740]
### 1. System Info

PyPOTS version: 0.17
Python version: 3.11.11

### 2. Information

- [ ] The official example scripts
- [x] My own created scripts

### 3. Reproduction

1. Use a 10 fold cross-validation on the entire dataset.
2. Initialize a PyPOTS classification model (I used SAITS). Set `patience=None`, `model_saving_strategy=None`.
3. Train the model by calling `model.fit(train_set, test_set)`.
4. Repeat steps above but do not pass the test set in fitting the model.
5. Compare the performance between the two runs.

### 4. Expected behavior

When patience is set to None, the early stopping is disabled, so the final model should be trained for the number of epochs specified in its initialization. In this case, when doing a cross-validation, supplying the test set as the validation set in `model.fit()` should only be used for information purpose and should not influence any part of training.

However, I observed that the model performed much better on the test set when supplying the test set in `model.fit()`, which means PyPOTS used the best model according to the validation loss from early epochs as the final model. This caused data leakage.

To summarize, I think there should be a mode to support cross-validation without an internal validation split within the training set. In this case, supplying the test set in `model.fit()` should have no effect on training. Originally I expected setting patience to None will achieve this, but only found out it was not this case.

### 5. Your contribution

Potential code to check: https://github.com/WenjieDu/PyPOTS/blob/main/pypots/base.py

1. Patience is set to -1 if it is None during model initialization.
2. However, best model is still being updated every epoch without checking initial value of patience - this should only be done for early stopping (i.e. patience >= 0).

**Comment by WenjieDu** [comment]
Hi Yufei, many thanks for catching this bug! Will you be interested in making a PR to fix it? This could be your 1st code contribution to PyPOTS ;-)

**Comment by mdehsan873** [comment]
Hi @WenjieDu can you please review the PR https://github.com/WenjieDu/PyPOTS/pull/744 ?

**Comment by WenjieDu** [comment]
Hi Yufei, "When patience is set to None, the early stopping is disabled, so the final model should be trained for the number of epochs specified in its initialization.", this is fine. But now that `val_set` is given, PyPOTS still should do model selection according to its performance on the validation set, and it will run out of all epochs. Regarding your expectation, as long as you want to validate the model but ignore the model selection step, you can set `model_saving_strategy` to "all" and keep `patience` as None, then you will get all saved model files at each epoch.

**Comment by WenjieDu** [comment]
Thank you Mohammad @mdehsan873 for making the PR. Please take a look at my explanation above. Even if we set patience as None, it doesn't mean we always want the model from the latest epoch.

**Comment by mdehsan873** [comment]
@WenjieDu That means it's not a bug, right?

**Comment by WenjieDu** [comment]
Yep, at least from my point of view so far. Let's see if my suggestion above can help Yufei get what he wants. 

BTW, will you be interested in helping with other issues ;-)

**Comment by mdehsan873** [comment]
@WenjieDu Yes
